### PR TITLE
Change LocalAuth to use Biometric API

### PR DESCRIPTION
**Pull request by mehmetf**
## Description

Migrates the LocalAuth plugin on Android to Biometric API for showing the fingerprint dialog. This will allow us to implement more biometric authentication schemes (and others such as PIN) in the future.

## Related Issues
https://github.com/flutter/flutter/issues/18690

## Breaking Change

This is a breaking change because it requires Flutter applications on Android to extend a FragmentActivity rather than a regular Activity. The fix is to use `FlutterFragmentActivity` instead of `FlutterActivity`. The y version has been bumped due to this.

### CODE REVIEW

**Comment by mehmetf** [comment]
@jiajiabingcheng Please take a look at this as well. It would be good if you copied this to Leafy's codebase and gave it a try.

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java [hunk 1]
```diff
@@ -147,99 +123,53 @@ private void stop(boolean success) {
...
-   * If the activity is paused or stopped, we have to stop listening for fingerprint. Otherwise,
-   * user can still interact with fp reader in the background.. Sigh..
-   */
   @Override
-  public void onActivityPaused(Activity activity) {
-    if (call.argument("stickyAuth")) {
-      pause();
-    } else {
+  public void onAuthenticationError(int errorCode, CharSequence errString) {
+    if (!activityPaused || !isAuthSticky) {
```
Comment (mklim) [inline packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java]: optional, subjective: The negatives combined with the logical or makes this kind of hard to read. Can we rephrase this?

```java
if (activityPaused && isAuthSticky) {
  return;
}

// Either the authentication got cancelled by user or we are not interested
// in sticky auth, so return failure.
stop(false);
```

Comment (mehmetf) [reply packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java]: Good suggestion. Applied.

**Comment by awazgyawali** [comment]
It seems that we have to change the Activity manually, shouldn't this be mentioned on the readme?

**Pull request merged**